### PR TITLE
[kueuectl] Pass-through commands.

### LINKED ISSUE #2180: [kueuectl] Add Pass-through commands
**What would you like to be added**:
Kueuectl Pass-through commands as described in  https://github.com/kubernetes-sigs/kueue/tree/main/keps/2076-kueuectl#pass-through.

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [x] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Add pass-through commands support in `kueuectl`. 

#### Which issue(s) this PR fixes:

Fixes #2180

#### Does this PR introduce a user-facing change?

```release-note
Added pass-through commands support in `kueuectl` for `get`, `describe`, `edit`, `patch` and `delete`.
```

### CODE REVIEW

File: test/integration/kueuectl/passthrough_test.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, key, obj)
+			}, util.Timeout, util.Interval).Should(testing.BeNotFoundError())
+		},
+		ginkgo.Entry("Workload", "workload", true, makePassThroughWorkload, "{.spec.active}", "'true'", `{"spec":{"active":false}}`, "'false'"),
+		ginkgo.Entry("Workload(short)", "wl", true, makePassThroughWorkload, "{.spec.active}", "'true'", `{"spec":{"active":false}}`, "'false'"),
+		ginkgo.Entry("Cluster Queue", "clusterqueue", false, makePassThroughCQ, "{.spec.stopPolicy}", "'None'", `{"spec":{"stopPolicy":"Hold"}}`, "'Hold'"),
+		ginkgo.Entry("Cluster Queue(short)", "cq", false, makePassThroughCQ, "{.spec.stopPolicy}", "'None'", `{"spec":{"stopPolicy":"Hold"}}`, "'Hold'"),
+		ginkgo.Entry("Local Queue", "localqueue", true, makePassThroughLQ, "{.spec.clusterQueue}", "'queue'", ``, ""),
+		ginkgo.Entry("Local Queue (short)", "lq", true, makePassThroughLQ, "{.spec.clusterQueue}", "'queue'", ``, ""),
```
Comment (alculquicondor) [inline test/integration/kueuectl/passthrough_test.go:132]: I'm happy with testing just one object, not to bloat the tests too much.

**Review by alculquicondor — COMMENTED** [review]
/approve
/hold to reduce integration tests.

File: test/integration/kueuectl/passthrough_test.go (hunk 1 above)
Comment (trasc) [reply test/integration/kueuectl/passthrough_test.go]: Done 

/unhold

File: apis/kueue/v1beta1/clusterqueue_types.go [hunk 2]
```diff
@@ -486,7 +486,7 @@ type FairSharing struct {
 // +genclient:nonNamespaced
 // +kubebuilder:object:root=true
 // +kubebuilder:storageversion
-// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:resource:scope=Cluster,shortName={cq}
```
Comment (tenzen-y) [inline apis/kueue/v1beta1/clusterqueue_types.go:489]: Actually, we dropped this shortName here: https://github.com/kubernetes-sigs/kueue/pull/604

So, I'm curious about whether we should re-enable this short name.
@alculquicondor  WDYT?

Comment (trasc) [reply apis/kueue/v1beta1/clusterqueue_types.go:489]: Do you remember the reason for that? 

In theory `kueuectl` can hack the args that are actually passed to `kubectl` in pass through commands , the problem is that `get cq` will end up working with `kueuectl` and create confusion when it comes to UX.

/cc @mwielgus

Comment (tenzen-y) [reply apis/kueue/v1beta1/clusterqueue_types.go:489]: Actually, I don't remember the reason...

Comment (alculquicondor) [reply apis/kueue/v1beta1/clusterqueue_types.go:489]: I just considered them too small and to potentially run into conflicts with CRDs from different projects.

But maybe it's worth, given a trend for users to use kueue without being too familiar with kubernetes, so I wouldn't expect many conflicts.

**Comment by alculquicondor** [comment]
/kind feature
/release-note-edit
```release-note
CLI: Added pass-through commands support in `kubectl-kueue` for `get`, `describe`, `edit`, `patch` and `delete`.
```

**Pull request merged**